use subtest for table units (pkg/scheduler)

xchapter7x opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:
1. If this is your first time, read our contributor guidelines https://git.k8s.io/community/contributors/guide#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
3. Follow the instructions for writing a release note: https://git.k8s.io/community/contributors/guide/release-notes.md
4. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->
**What this PR does / why we need it**: Update scheduler's unit table tests to use subtest

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

**Special notes for your reviewer**:
breaks up PR: https://github.com/kubernetes/kubernetes/pull/63281
/ref #63267

**Release note**:
<!--  Write your release note:
1. Enter your extended release note in the below block. If the PR requires additional action from users switching to the new release, include the string "action required".
2. If no release note is required, just write "NONE".
-->
```release-note
This PR will leverage subtests on the existing table tests for the scheduler units.
Some refactoring of error/status messages and functions to align with new approach.

```

misterikkit commented on pkg/scheduler/scheduler_test.go:
```diff
@@ -177,79 +183,113 @@ func TestScheduler(t *testing.T) {
-		}
-		if e, a := item.expectError, gotError; !reflect.DeepEqual(e, a) {
-			t.Errorf("%v: error: wanted %v, got %v", i, e, a)
+	s, _ := NewFromConfigurator(configurator, nil...)
```
Another weird one. `nil...` is the same as not passing any args after configurator. (don't fix in this PR. I'm taking notes though.)

misterikkit commented on pkg/scheduler/scheduler_test.go:
```diff
@@ -675,15 +742,17 @@ func TestSchedulerWithVolumeBinding(t *testing.T) {
 	}{
-		"all-bound": {
+		{
+			name: "all-bound",
```
It's okay to use spaces in test names. t.Run will replace them with underscore.

misterikkit commented on pkg/scheduler/scheduler_test.go:
```diff
@@ -694,7 +763,8 @@ func TestSchedulerWithVolumeBinding(t *testing.T) {
 		},
-		"bound,invalid-pv-affinity": {
+		{
+			name: "bound,invalid-pv-affinity",
```
The commas could probably be replaced with slashes in these test names. (I'm assuming you already know how `-run` interacts with subtests and subsubtests.)

misterikkit commented:
/ok-to-test

xchapter7x commented:
/test pull-kubernetes-integration

xchapter7x commented:
/test pull-kubernetes-integration
/test pull-kubernetes-e2e-gce

xchapter7x commented:
/assign @davidopp

xchapter7x commented:
/test pull-kubernetes-e2e-kops-aws
/test pull-kubernetes-bazel-build
/test pull-kubernetes-e2e-gce-device-plugin-gpu

xchapter7x commented:
/assign @misterikkit 
this PR should be in a good spot if you have a moment to take a look. thanks.

misterikkit commented on pkg/scheduler/scheduler_test.go:
```diff
@@ -177,79 +183,113 @@ func TestScheduler(t *testing.T) {
+}
+
+func testScheduler(
+	injectBindError error,
```
As we discussed in https://github.com/kubernetes/kubernetes/pull/63665#issuecomment-393884049, let's keep this inline for now.

misterikkit commented on pkg/scheduler/scheduler_test.go:
```diff
@@ -381,53 +421,80 @@ func TestSchedulerErrorWithLongBinding(t *testing.T) {
 	conflictPod := podWithPort("bar", "", 8080)
 	pods := map[string]*v1.Pod{firstPod.Name: firstPod, conflictPod.Name: conflictPod}
 	for _, test := range []struct {
+		Name            string
```
lowercase name

xchapter7x committed:
use subtest for table units

  apply consistent format to name strings

  - #63661 (comment)
  - #63661 (comment)

inline subtest logic

  https://github.com/kubernetes/kubernetes/pull/63661/files#r192540031

remove duplicate messaging in subtest errors

xchapter7x commented:
/test pull-kubernetes-e2e-gce-100-performance

misterikkit commented:
/lgtm

bsalamat reviewed (approved):
/approve

Thanks, @xchapter7x!
